src/libs/actions/TaxRate.ts

```diff
@@ -119,10 +144,300 @@ function clearTaxRateError(policyID: string, taxID: string, pendingAction?: Pend
+                                errors: ErrorUtils.getMicroSecondOnyxError('workspace.taxes.errors.genericFailureMessage'),
+                            };
+                            return acc;
+                        }, {} as TaxRateDeleteMap),
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters = {
+        policyID,
+        taxNames: JSON.stringify(taxesToDelete.map((taxID) => policyTaxRates[taxID].name)),
+    } as DeletePolicyTaxesParams;
+
+    API.write(WRITE_COMMANDS.DELETE_POLICY_TAXES, parameters, onyxData);
+}
+
+/**
+ * Rename policy tax
+ */
+function updatePolicyTaxValue(policyID: string, taxID: string, taxValue: number) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+    const originalTaxRate = {...policy?.taxRates?.taxes[taxID]};
+    const stringTaxValue = `${taxValue}%`;
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxID]: {
+                                value: stringTaxValue,
+                                pendingFields: {value: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
+                                errorFields: {value: null},
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxID]: {value: stringTaxValue, pendingFields: {value: null}, errorFields: {value: null}},
+                        },
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxID]: {
+                                value: originalTaxRate.value,
+                                pendingFields: {value: null},
+                                errorFields: {value: ErrorUtils.getMicroSecondOnyxError('workspace.taxes.errors.genericFailureMessage')},
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+    };
+
+    if (!originalTaxRate.name) {
+        throw new Error('Tax rate name not found');
+    }
+
+    const parameters = {
+        policyID,
+        taxCode: taxID,
+        taxAmount: Number(taxValue),
+    } as UpdatePolicyTaxValueParams;
```

Comment:
why is this assertion needed?

Reply:
Why do we need it?

Reply:
We don't do it for other actions

Reply:
I mean it's not necessary, but it's a bit stricter type. If needed, I can remove it.